diff --git a/android/build.gradle b/android/build.gradle
index 6b73a69d82ae9b04dd41e22b2d115019cf7eae14..24747e22c076d2b50be6e89511c26846fa8a0ddb 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -203,7 +203,8 @@ android {
                         "-DREANIMATED_PROFILING=${REANIMATED_PROFILING}",
                         "-DREANIMATED_VERSION=${REANIMATED_VERSION}",
                         "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
-                        "-DREANIMATED_FEATURE_FLAGS=${REANIMATED_FEATURE_FLAGS}"
+                        "-DREANIMATED_FEATURE_FLAGS=${REANIMATED_FEATURE_FLAGS}",
+                        "-DCMAKE_OBJECT_PATH_MAX=512"
                 abiFilters (*reactNativeArchitectures())
                 targets("reanimated")
             }
